R/get_fields.R

```diff
@@ -57,11 +58,36 @@ get_fields <- function(es_host
     mappingDT <- .flatten_mapping(mapping = resultContent)
     
     ##################### get aliases for index names #########################
-    aliasDT <- .get_aliases(es_host = es_host)
-    if (!is.null(aliasDT)) {
-        lookup <- aliasDT[['alias']]
-        names(lookup) <- aliasDT[['index']]
-        mappingDT[index %in% names(lookup), index := lookup[index]]
+    rawAliasDT <- .get_aliases(es_host = es_host)
+    if (!is.null(rawAliasDT)) {
+        
+        log_info("Replacing index names with aliases")
+        
+        # duplicate the mapping results for every alias. Idea is that you should be able to rely 
+        # on the results of get_fields to programmatically generate queries, and you should have a
+        # preference for hitting aliases over straight-up index name
+        aliasDT <- data.table::rbindlist(
+            purrr::map2(
+                .x = rawAliasDT[["index"]]
+                , .y = rawAliasDT[["alias"]]
+                , .f = function(idx_name, alias_name, mappingDT){
+                    tmpDT <- mappingDT[index == idx_name]
+                    tmpDT[, index := alias_name]
+                    return(tmpDT)
+                }
+                , mappingDT = mappingDT
+            )
+            , fill = TRUE
+        )
```

Comment:
I think there's a better way you could do this with a data.table merge but these DTs are small anyway so it probably doesn't matter

Reply:
yeah they're pretty small and this makes it really explicit. This is an EDA function so super-super fast performance is less important than correctness, IMHO